pyccel/codegen/printing/ccode.py

```diff
@@ -953,8 +954,12 @@ def _print_FunctionDef(self, expr):
             self._additional_args.append(expr.results)
         body  = self._print(expr.body)
         decs  = [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.local_vars]
-        if len(expr.results) <= 1 :
-            decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
+        if len(expr.results) == 1 :
+            if isinstance(expr.results[0], Variable):
+                if expr.results[0] not in self._unnecessary_declaration:
+                    decs += [Declare(expr.results[0].dtype, expr.results[0])]
+            elif isinstance(expr.results[0], FunctionAddress):
+                decs += [FuncAddressDeclare(expr.results[0])]
```

Comment:
We don't add function's docstring for making pylint or Codacy happy, we are doing it for anyone who wants to read our code, the same for a line comment.
It may be easy to read your code now, but it won't be in the future.
so please try to add comments on your code.

Reply:
by this, we will have a code like the header files of an OS (e.g [this](https://opensource.apple.com/source/cctools/cctools-795/include/mach-o/loader.h)), if you really want to understand the code just ask me.

Reply:
Documentation is always important if the purpose of the code is not clear from reading it. You don't need a comment per line, but a comment per idea is usually a good idea

>  if you really want to understand the code just ask me.

This is *always* a bad answer as you have no idea when someone will have questions about this. If someone has a question in 3 years time then you may not remember any more and you may not be working on pyccel anymore. In the worst case we may not even have a way to contact you any more. This is why clear, well documented code is *very* important, especially in a big project

Reply:
the thing is just how can i tell is the code is readable for someone, So by asking me i can tell that and write some clarifications.

> a comment per idea

i do agree with that, actually [this](https://github.com/pyccel/pyccel/pull/683/commits/700bc7c3452545a77cc21e9135e229c971b6972e) what i do when i wrote a line that not answering the question  "why"  from my perspective.